Applied-Upstream: commit:9b775d613ad0c324db628ed5a32d3fccc90f82d6
Author: Joshua Strobl <JoshStrobl@users.noreply.github.com>
Last-Update: 2024-02-11
Description: [PATCH] fix: workspace applet window icon click not performing
 workspace switch (#525)

---
 src/panel/applets/workspaces/WindowIcon.vala | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/src/panel/applets/workspaces/WindowIcon.vala b/src/panel/applets/workspaces/WindowIcon.vala
index 8995be1b..a9efc023 100644
--- a/src/panel/applets/workspaces/WindowIcon.vala
+++ b/src/panel/applets/workspaces/WindowIcon.vala
@@ -55,14 +55,10 @@ namespace Workspaces {
 		public override bool button_release_event(Gdk.EventButton event) {
 			if (event.button != 1) return Gdk.EVENT_STOP;
 
-			var workspace = WorkspacesApplet.workspace_group.get_active_workspace();
-			if (workspace != null && workspace == window.get_workspace()) {
-				try {
-					window.activate(event.time);
-				} catch (Error e) {
-					warning("Failed to activate window: %s", e.message);
-				}
-				return Gdk.EVENT_STOP;
+			try {
+				window.activate(event.time);
+			} catch (Error e) {
+				warning("Failed to activate window: %s", e.message);
 			}
 			return Gdk.EVENT_STOP;
 		}
-- 
2.43.0

